ProcessList.c

```diff
@@ -557,9 +557,24 @@ void ProcessList_rebuildPanel(ProcessList* this) {
 
       if ( (!p->show)
          || (this->userId != (uid_t) -1 && (p->st_uid != this->userId))
-         || (incFilter && !(String_contains_i(Process_getCommand(p), incFilter)))
-         || (this->pidMatchList && !Hashtable_get(this->pidMatchList, p->tgid)) )
+         || (incFilter && !(String_contains_i(Process_getCommand(p), incFilter))) ) {
          continue;
+      }
+
+      if (this->pidMatchList && !Hashtable_get(this->pidMatchList, p->tgid)) {
+         /* Follow processes parents to find matches to process match list */
+         Process* currProc = p;
+         bool found = Hashtable_get(this->pidMatchList, currProc->tgid);
+         while (!Process_isChildOf(currProc, 0) && !found) {
```

Comment:
In theory it's possible for two processes in one snapshot to list each other as parents in case of really unfortunate PID reuse race. Not sure whether that's worth guarding against - it's really unlikely, but can hang htop completely with an implementation like this.

Reply:
On the other hand, I'm not really sure whether other parts of code are safe against that :-)

Reply:
Good catch and definitively worth investigating.

As reading the process list can take a significant amount of time, this race is a real chance on a VERY busy system. so guarding against it is adviseable, but should be done with minimum performance impact, as this is a hot code path.

Reply:
What are your thoughts on having a max depth that if we hit (lets just say) 10  iterations, we exit the while loop? Maybe having it as an option passed in such as '--pid <pid> --max-child-depth 10'

It doesnt address the possibility of such a pid reuse error affecting the tree building process, but it would be cheap in cpu/memory usage, and avoid getting stuck in the loop.